### PR TITLE
[cinder]: add reset and force delete actions to volumes and snapshots

### LINKED ISSUE #2695: [cinder]: implement reset status action to volumes and snapshots
https://docs.openstack.org/api-ref/block-storage/v3/?expanded=reset-a-volume-s-statuses-detail#reset-a-volume-s-statuses
https://docs.openstack.org/api-ref/block-storage/v3/?expanded=reset-a-snapshot-s-status-detail#reset-a-snapshot-s-status

### PR DESCRIPTION
**Pull request by kayrus**
Resolves #2695

### CODE REVIEW

File: openstack/blockstorage/extensions/volumeactions/requests.go [hunk 1]
```diff
@@ -418,3 +418,43 @@ func ReImage(client *gophercloud.ServiceClient, id string, opts ReImageOpts) (r
 	_, r.Header, r.Err = gophercloud.ParseResponse(resp, err)
 	return
 }
+
+// ResetStatusOptsBuilder allows extensions to add additional parameters to the
+// ResetStatus request.
+type ResetStatusOptsBuilder interface {
+	ToSnapshotResetStatusMap() (map[string]interface{}, error)
```
Comment (mandre) [inline openstack/blockstorage/extensions/volumeactions/requests.go:425]: Should be `ToResetStatusMap()`.

File: openstack/blockstorage/extensions/volumeactions/requests.go
Comment (mandre) [inline openstack/blockstorage/extensions/volumeactions/requests.go:1]: There are a few references to snapshots in this file, most probably due to copy & paste.

**Review by mandre — COMMENTED** [review]
Apart from the `snapshot` bits sneaking into the volume action this is good.

File: openstack/blockstorage/extensions/volumeactions/requests.go (hunk 1 above)
Comment (kayrus) [reply openstack/blockstorage/extensions/volumeactions/requests.go]: Nice catch. Fixed.

File: openstack/blockstorage/extensions/volumeactions/requests.go
Comment (kayrus) [reply openstack/blockstorage/extensions/volumeactions/requests.go:1]: Fixed.

**Review by mandre — APPROVED** [review]

**Pull request merged**